### ISSUE TITLE
[4.0] schema checker and mysql 8

### ISSUE THREAD

**Issue by alikon** [joomla/joomla-cms#28367]
### Steps to reproduce the issue
with mysql 8
go to System -> Database

### Expected result

All database table structures are up to date.
no problems

### Actual result

![modified](https://user-images.githubusercontent.com/181681/76739124-171b1080-676c-11ea-89a5-1d8b99967366.png)

cc  @richard67

**Comment by richard67** [comment]
Confirmed. I have that, too, on my MySQL 8. The reason is that a "SHOW COLUMNS" statement, which is what is used by the schema checker, returns e.g. "int" or "int unsigned" in the "type" column, where a MySQL 5.7 shows e.g. "int(11)" or "int(10) unsigned" or "int(5)" or whatever display size was given. I.e. the display size is not included anymore in the "type" value for integers in MySQL 8.

As one can read here [https://dev.mysql.com/doc/refman/8.0/en/numeric-type-attributes.html](https://dev.mysql.com/doc/refman/8.0/en/numeric-type-attributes.html), this display size is only relevant for padding with zeros, which is deprecated anyway, as described there. It does not change anything on the value range of that particular integer field, and it does not change anything on the storage space it needs.

I have stumpled at the same issue yesterday night and have already prepared a fix.

@alikon Thanks for reproducing and creating this issue.

Will make PR later today. The PR will make the schema checker for MySQL accept both, "int" and "int unsigned" with and without display size, so it works for both MySQL 8 and previous. That will mean the schema checker will simply ignore the display size for integer columns.
